ddtrace/contrib/django/patch.py

```diff
@@ -401,60 +400,64 @@ def traced_get_response(django, pin, func, instance, args, kwargs):
             # Note: this call must be done after the function call because
             # some attributes (like `user`) are added to the request through
             # the middleware chain
-            _set_request_tags(django, span, request)
-
-            if response:
-                status = response.status_code
-                span.set_tag("django.response.class", func_name(response))
-                if hasattr(response, "template_name"):
-                    # template_name is a bit of a misnomer, as it could be any of:
-                    # a list of strings, a tuple of strings, a single string, or an instance of Template
-                    # for more detail, see:
-                    # https://docs.djangoproject.com/en/3.0/ref/template-response/#django.template.response.SimpleTemplateResponse.template_name
-                    template = response.template_name
-
-                    if isinstance(template, six.string_types):
-                        template_names = [template]
-                    elif isinstance(
-                        template,
-                        (
-                            list,
-                            tuple,
-                        ),
-                    ):
-                        template_names = template
-                    elif hasattr(template, "template"):
-                        # ^ checking by attribute here because
-                        # django backend implementations don't have a common base
-                        # `.template` is also the most consistent across django versions
-                        template_names = [template.template.name]
+            try:
+                _set_request_tags(django, span, request)
```

Comment:
Just wanted to double-check that we must wrap this whole block in `try`...`except` here, since the error message is specific to just this line and as far as I can see, all calls to `getattr` re. `user` are done in `_set_request_tags`.

Reply:
Hi @P403n1x87, the `try ... except` was originally meant to be contained within `_set_request_tags`, but we quickly found that similar `ImproperlyConfigured` errors from Django could be propagated into our tracer here when the `response.template` objects are also accessed within this block. I thought a defensive approach with the `try...except` blanket over this block would be more appropriate. What do you think?

Reply:
If I am correct in thinking that the trace is salvaged despite the error, I would try and set as much available info on it as possible. Hence, I would wrap just the part that is known to fail and log the appropriate error. So, in this case, I would also wrap the other part that is known to potentially fail and that can be handled within another `try...except` so that when we recover we keep adding stuff till the end.

Reply:
From the latest django framework test suite run, it looks like the template tags setting does not actually raise any internal django errors - i.e. this has been handled already by moving the exception handling to be inside _set_requests_tags() as was done initially. I'll revert this change and leave the larger code block as it was originally.